distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/CometDClient.java

```diff
@@ -0,0 +1,199 @@
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.common.test;
+
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import javax.net.ssl.HostnameVerifier;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+import org.cometd.bayeux.Channel;
+import org.cometd.bayeux.Message;
+import org.cometd.bayeux.client.ClientSessionChannel;
+import org.cometd.client.BayeuxClient;
+import org.cometd.client.transport.ClientTransport;
+import org.cometd.client.transport.LongPollingTransport;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.slf4j.LoggerFactory;
+import org.slf4j.ext.XLogger;
+import org.springframework.util.CollectionUtils;
+
+public class CometDClient {
+
+    private static final XLogger LOGGER = new XLogger(LoggerFactory.getLogger(CometDClient.class));
+    
+    private static final long TIMEOUT = TimeUnit.SECONDS.toMillis(60);
+    
+    private final BayeuxClient bayeuxClient;
+
+    private final HttpClient httpClient;
+
+    private final List<MessageListener> messageListeners = new ArrayList<>();
+
+    public CometDClient(String url) throws Exception {
+        SslContextFactory sslContextFactory = new SslContextFactory(true);
+        httpClient = new HttpClient(sslContextFactory);
+        doTrustAllCertificates();
+        ClientTransport transport = new LongPollingTransport(new HashMap<>(), httpClient);
+        bayeuxClient = new BayeuxClient(url, transport);
+    }
+
+    public boolean start() throws Exception {
+        httpClient.start();
+        LOGGER.debug("HTTP client started: {}", httpClient.isStarted());
+        MessageListener handshakeListener = new MessageListener(Channel.META_HANDSHAKE);
+        bayeuxClient.getChannel(Channel.META_HANDSHAKE).addListener(handshakeListener);
+        bayeuxClient.handshake();
+        boolean connected = bayeuxClient.waitFor(TIMEOUT, BayeuxClient.State.CONNECTED);
+        return connected;
```

Comment:
As a client, it's easy to forget to check the flag and then never realize that the connection never got setup. Should we throw an exception instead if the connection fails or can you think of a reason when returning a `boolean` would be useful?